### DISCUSSION TITLE
Fireshell Discussions

### DISCUSSION THREAD

**Discussion by AayushK47** [AayushK47/fireshell#18] [General]
This is a general discussion. Anyone can participate.

**Comment by NimperX** [comment]
Hello @AayushK47 , What exactly need to be done? Rewrite the app in typescript, is it?

**Reply by AayushK47** [reply]
Hi there. Yes I am currently rewriting the app in ts, mainly because the current code is not really readable and also has some vulnerabilities. You can start by cloning the repo and start using the app, understand how it works. Then i will explain how the rewrite is being structured.
